src/process.rs

```diff
@@ -39,21 +39,67 @@ impl Drop for TemporaryProcess {
     }
 }
 
-pub struct LaunchOptions<'a> {
+pub struct LaunchOptions {
     pub headless: bool,
     pub port: Option<u16>,
-    pub path: &'a str,
+    pub path: std::path::PathBuf,
 }
 
-impl<'a> Default for LaunchOptions<'a> {
-    fn default() -> Self {
-        LaunchOptions {
+impl LaunchOptions {
+    pub fn default_executable() -> Option<std::path::PathBuf> {
+        // TODO BSDs/Unixes are the same?
+        #[cfg(target_os = "linux")]
+        {
+            // TODO Look at $BROWSER and if it points to a chrome binary
+            // $BROWSER may also provide default arguments, which we may
+            // or may not override later on.
+
+            // TODO More paths!?
+            let default_paths = &["/usr/bin/google-chrome-stable", "/usr/bin/chromium-browser"][..];
+            for path in default_paths {
+                if std::path::Path::new(path).exists() {
+                    return Some(path.into());
+                }
+            }
+        }
+
+        #[cfg(target_os = "macos")]
+        {
+            let default_paths =
+                &["/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"][..];
+            for path in default_paths {
+                if std::path::Path::new(path).exists() {
+                    return Some(path.into());
+                }
+            }
+        }
+        // TODO Windows
+        // Check HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\App Paths\chrome.exe
+        None
+    }
+
+    pub fn default() -> Option<Self> {
```

Comment:
How would you feel about this being a `Result` instead? If someone calls `LaunchOptions::default()` and gets back `None`, they might not understand why, or what they need to do to fix it.

Reply:
And it's also let construction be like this:

`Browser::new(LaunchOptions::default()?)?`

Because you can't use the `?` for `Result`s and `Option`s in the same function, right?

Reply:
I've added a default constructor to `Browser`, so the `Browser::new(LaunchOptions::default()?)?` can be written as `Browser::default()?`